main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DuplicateRouteNameValidator.java

```diff
@@ -45,50 +46,50 @@
 
   @Override
   public void validate(NoticeContainer noticeContainer) {
-    final Map<String, GtfsRoute> routeByLongName = new HashMap<>(routeTable.entityCount());
-    final Map<String, GtfsRoute> routeByShortName = new HashMap<>(routeTable.entityCount());
-    final Map<String, GtfsRoute> routeByShortAndLongName = new HashMap<>(routeTable.entityCount());
+    final Map<Integer, GtfsRoute> routeByLongName = new HashMap<>(routeTable.entityCount());
+    final Map<Integer, GtfsRoute> routeByShortName = new HashMap<>(routeTable.entityCount());
+    final Map<Integer, GtfsRoute> routeByShortAndLongName = new HashMap<>(routeTable.entityCount());
     routeTable
         .getEntities()
         .forEach(
             route -> {
               if (route.hasRouteShortName() && route.hasRouteLongName()) {
-                if (routeByShortAndLongName.containsKey(getShortAndLongNameKey(route))) {
+                if (routeByShortAndLongName.containsKey(getShortAndLongNameRouteTypeHash(route))) {
                   noticeContainer.addValidationNotice(
                       new DuplicateRouteNameNotice(
                           "route_short_name and route_long_name",
                           route.csvRowNumber(),
                           route.routeId()));
                   return;
                 } else {
-                  routeByShortAndLongName.put(getShortAndLongNameKey(route), route);
+                  routeByShortAndLongName.put(getShortAndLongNameRouteTypeHash(route), route);
                 }
               }
               if (route.hasRouteLongName()) {
-                if (routeByLongName.containsKey(getRouteLongNameKey(route))) {
+                if (routeByLongName.containsKey(getRouteLongNameRouteTypeHash(route))) {
```

Comment:
I think this would be easier to read, and also avoids looking up the hash twice and computing the key twice, if you assign to local variables here, like:

```java
                int key = getRouteLongNameKey(route);
                GtfsRoute otherRoute = routeByLongName.get(key);
                if (otherRoute != null) {
                  if (areRoutesFromSameAgency(route.agencyId(), otherRoute.agencyId())) {
                    noticeContainer.addValidationNotice(
                        new DuplicateRouteNameNotice(
                            "route_long_name", route.csvRowNumber(), route.routeId()));
                  }
                  return;
                } else {
                  routeByLongName.put(key, route);
                }
```

Same for route short name and route short and long name IF statements.